### ISSUE TITLE
Build fails: "end-of-line within string"

### ISSUE THREAD

**Issue by Stiivi** [ihhub/fheroes2#8237]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

macOS

### Describe the bug

The build fails with:

```
sed -e 'y/äëïöőüűÄËÏŐÖÜŰ/aeioouuAEIOOUU/' lt.po | iconv -f utf-8 -t ascii//TRANSLIT | msgfmt - -o lt.mo
<stdin>:3865: end-of-line within string
iconv: warning: invalid characters: 702
msgfmt: found 1 fatal error
make[1]: *** [Makefile:61: lt.mo] Error 1
make[1]: Leaving directory '/Users/stefan/Developer/Sources/fheroes2/files/lang'
make: *** [Makefile:39: all] Error 2
```

Broken commit: 5dfd271c0fb41e00927252699abea36aba12aafb

Last working commit (no error): 4174e71dd640ccda6d808876c05c24f284efaa00

### Save file

(no save file)

**Comment by Stiivi** [comment]
No changes on my side. Just did a clean git clone of 92b39d82d9fcd998a8775acdcc919a4973d03c5f into a new location, got the same error.

Attached is whole checout+build session with some git info at the end.

[fheroes2-build-session.txt](https://github.com/ihhub/fheroes2/files/13850808/fheroes2-build-session.txt)

**Comment by oleg-derevenetz** [comment]
I cannot reproduce this issue on my macOS with neither make nor CMake. Our CI cannot reproduce this issue as well. What is your locale? Is it UTF-8 locale? What will be the output of `env | grep LANG` command in your console?

**Comment by Stiivi** [comment]
`env | grep LANG` gives no results - no env variable set. There is only one env. locale-related variable set: `LC_CTYPE=UTF-8`.

**Comment by ihhub** [comment]
Hi @Stiivi , do you use macOS 14?

**Comment by ihhub** [comment]
I believe [this Homebrew pull request](https://github.com/Homebrew/homebrew-core/pull/158820) is facing the same problem.

**Comment by Stiivi** [comment]
Using MacOS Sonoma 14.2.1.

The lined Homebrew issue seems to be exhibiting the same issue:

```
<stdin>:3830: keyword "e" unknown
```

When we look at the file triggering the problem in fheroes after running `sed -e 'y/äëïöőüűÄËÏŐÖÜŰ/aeioouuAEIOOUU/' files/lang/lt.po | iconv -f utf-8 -t ascii//TRANSLIT` we get a result with the following content (line numbers added):

```
3860
3861 #, fuzzy
3862 msgid "Campaign Difficulty"
3863 msgstr ""
3864 "Zemelapis\ne
3865 "Sud^W^Atingumas"
```

see line 3864 where `e` was added instead of `"`. Or rather, after few experiments, it is adding `e` instead the last character of that line, no matter what the original character is.

Maybe a bug in `iconv` on MacOS 14?

**Comment by Stiivi** [comment]
I found a dirty workaround:

Insert a dummy comment with at least 6 characters after `#` before line 3861.

```diff
diff --git a/files/lang/lt.po b/files/lang/lt.po
index 2aa563fc..3c1ba0b3 100644
--- a/files/lang/lt.po
+++ b/files/lang/lt.po
@@ -3858,6 +3858,7 @@ msgstr ""
 msgid "Hard"
 msgstr ""
 
+# -----
 #, fuzzy
 msgid "Campaign Difficulty"
 msgstr ""
```

... it seems to reset some broken counter(?) in iconv and the error triggering line gets converted correctly.

**Comment by ihhub** [comment]
Hi @Stiivi , it is only macOS 14 issues as I see that other builds for Homebrew are successful.

@oleg-derevenetz , I assume that you use older than macOS 14 system, don't you?

**Comment by oleg-derevenetz** [comment]
> I assume that you use older than macOS 14 system, don't you?

Yes.

**Comment by oleg-derevenetz** [comment]
[They say](https://github.com/fredowski/homebrew-pspp/issues/5#issuecomment-1793464558) Apple rewrote `iconv` in macOS 14 by removing the GPL-licensed parts. Apparently, they introduced several bugs at the same time.

**Comment by p-linnane** [comment]
macOS system `iconv` used to be based on GNU libiconv but now it's based on FreeBSD's. System `iconv` was a mess when the first release of macOS 14 came out. We had to add workarounds to get it working, but most were fixed in subsequent updates. Will see how we can handle this in Homebrew.

**Comment by ihhub** [comment]
I am suggesting to disable this line (add fuzzy comment). I am aware that this is a workaround but Lithuanian translation is not currently maintained and we unblock fheroes2 engine usage for macOS 14 users. What do you think @zenseii and @oleg-derevenetz ?

**Comment by oleg-derevenetz** [comment]
> I am suggesting to disable this line (add fuzzy comment). I am aware that this is a workaround but Lithuanian translation is not currently maintained and we unblock fheroes2 engine usage for macOS 14 users.

If it helps, then I see no reason not to.

**Comment by ihhub** [comment]
Hi @Stiivi , can you please make a pull request based on your suggestion?

**Comment by ihhub** [comment]
Fixed by #8278.
